### PR TITLE
docs: add reference to gateway-api and sort index

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolves: #624

image

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -15,6 +15,9 @@ standalone or Kubernetes-based application gateway.
    This project is under active development. Many, many features are not
    complete. We would love for you to :doc:`get involved<get_involved>`.
 
+   This project is now listed in official
```
Comment (danehans) [inline docs/index.rst:18]: s/in official/as an official/

IMO this information is better conveyed on the compatibility page. Thoughts @arkodg @LukeShu @skriss @AliceProxy?

File: docs/index.rst [hunk 2]
```diff
@@ -15,6 +15,9 @@ standalone or Kubernetes-based application gateway.
    This project is under active development. Many, many features are not
    complete. We would love for you to :doc:`get involved<get_involved>`.
 
+   This project is now listed in official 
+   `Gateway API implementations <https://gateway-api.sigs.k8s.io/implementations/#envoy-gateway>`_ .
```
Comment (danehans) [inline docs/index.rst:19]: s/implementations/implementation/

**Comment by arkodg** [comment]
imho if we we are linking back to the Gateway API docs, there should be a good reason why it matters to the user
* either we are encouraging them to check out other implementations :)
* or the Gateway API docs has stated the level of conformance/support which Envoy Gateway provides (currently `core` conformance) reassuring the user of feature support.
I think we should highlight the latter point, raised https://github.com/kubernetes-sigs/gateway-api/pull/1475 to highlight that EG passes `core` conformance tests and we should include that verbiage here as well.

**Comment by danehans** [comment]
@Xunzhuo do you mind resolving the review feedback or closing this PR?

**Comment by Xunzhuo** [comment]
Yes @danehans,  I think we can close this PR.

**Pull request closed without merging**